jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -545,7 +545,22 @@ def _plot_uncertainties(self):
                 self.figure.marks = list(self.figure.marks) + [error_line_mark]
 
     def set_plot_axes(self):
-        # Set axes labels for the spectrum viewer
+        # Set y axes labels for the spectrum viewer
+        if y_unit.is_equivalent(u.Jy):
+            flux_unit_type = "Flux density"
```

Comment:
Besides "surface brightness" that Kyle is working on, I think the rest could be inferred from `y_unit.physical_type`?

Reply:
re surface brightness, see https://github.com/astropy/astropy/pull/16032

Reply:
I can still do it this way, but some of the labels we want will differ from what is returned by `physical_type` ( for example, an equivalency u.W should make the axis label 'luminosity', and u.electron/u.s should be 'counts') so it might not actually be less lines